python/paddle/fluid/layers/nn.py

```diff
@@ -3978,3 +3979,53 @@ def _is_list_or_turple_(data):
         attrs={"out_h": out_h,
                "out_w": out_w})
     return out
+
+
+def gather(input, index):
+    """
+    **Gather Operator**
+
+    Out is obtained by gathering entries of the outer-most dimension 
+    of X indexed by Index and concatenate them together.
+
+    .. math::
+
+	Out = X[Index]
+
+
+    .. code-block:: text
+
+	* Case 1:
+
+    		 X = [[1, 2],
+         	      [3, 4],
+                      [5, 6]]
+
+                Index = [1, 2]
+
+                Then:
+
+                Out = [[3, 4],
+                       [5, 6]]
+
+    Args:
+        input (Variable): The source input with rank>1. 
```

Comment:
gather_op should work even if `rank == 1`, please consider to modify it.

Reply:
Thx. The op support `rank==1` already. I add a unitest for `rank==1`.